daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -323,16 +275,17 @@ object Speedy {
       if (!isFinal) {
         crash("toSValue: machine not final")
       } else {
-        ctrl match {
-          case SEValue(v) => v
-          case _ => crash("machine did not evaluate to a value")
-        }
+        ctrl_value
       }
 
     def print(count: Int) = {
       println(s"Step: $count")
       println("Control:")
-      println(s"  ${ctrl}")
+      if (ctrl_value != null) {
+        println(s"  ${ctrl_value}")
+      } else {
+        println(s"  ${ctrl_expr}")
+      }
```

Comment:
Is this supposed to stay in?

Reply:
What's wrong with it? Is was there before already. And it's in a function used only in debugging tools anyway.